src/librbd/LibrbdAdminSocketHook.cc

```diff
@@ -78,6 +91,13 @@ LibrbdAdminSocketHook::LibrbdAdminSocketHook(ImageCtx *ictx) :
   if (r == 0) {
     commands[command] = new InvalidateCacheCommand(ictx);
   }
+
+  command = "rbd aio cancel " + imagename;
```

Comment:
I'm not comfortable adding a "data corrupting" ASOK hook. Perhaps in combination w/ some "rbd/objecter data corrupting admin commands enabled = true"-style config override. Also, why not just create an admin command that is directly implemented within the ```Objecter``` to avoid any changes to librbd / librados?